### PR TITLE
Update notifications rules: make a sound for each notification

### PR DESCRIPTION
**Pull request by Claire1817**
## Type of change

- [X] Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content
#4632 
Align notification experience to web / ios client.
Notify with noise for each notification.

## Tests

- Step 1
- Step 2
- Step ...

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [X] Changes has been tested on an Android device or Android emulator with API 21
- [] UI change has been tested on both light and dark themes
- [] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [X] Pull request is based on the develop branch
- [X] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

**Comment by ouchadam** [comment]
some things to look out for when making changes to this area
- spam when using a connected device like a smart watch/fitness band https://github.com/vector-im/element-android/issues/1673
- double notifications when receiving [fastlane updates](https://github.com/vector-im/element-android/blob/develop/vector/src/gplay/java/im/vector/app/gplay/push/fcm/VectorFirebaseMessagingService.kt#L204) followed by a sync https://github.com/vector-im/element-android/issues/4152

Hopefully the combination of all fixes in this area might have solved these :crossed_fingers:

**Comment by ouchadam** [comment]
for #1673 - tested locally on a mi band 5 and can confirm there's no regression 

however #4152 returns, a way to test this locally is to apply a delay to https://github.com/vector-im/element-android/blob/develop/vector/src/gplay/java/im/vector/app/gplay/push/fcm/VectorFirebaseMessagingService.kt#L174

```kotlin
coroutineScope.launch {
    delay(5000)
    session.requireBackgroundSync()
}
```

to avoid double notifications we'll need to further refine the notifications to dynamically enable `alertOnce` if the notification is new or an update

**Comment by Claire1817** [comment]
@ouchadam I had tested the #4152 by adding the "delay" and i didn't reproduce the bug.

**Review by mnaturel — COMMENTED** [review]
After some tests with an emulator Pixel 3a Android 12, I frequently have 2 notification sounds per message. I don't know how notifications work on the project but it may be worth to check/add logs when `notify()` is called on the `NotificationManager` for a single message.

**Comment by Claire1817** [comment]
@mnaturel I didn't reproduce your bug with an emulator.
I had tested with long and short messages, good and bad network. 
Can you test it on device ?

**Comment by mnaturel** [comment]
> @mnaturel I didn't reproduce your bug with an emulator. I had tested with long and short messages, good and bad network. Can you test it on device ?

After some tests on real device (Nokia 7.2, Android 11) I could not reproduce the double notification bug. I suspect it may happen sometimes on very high connectivity.

**Review by mnaturel — APPROVED** [review]

**Comment by ouchadam** [comment]
To clarify the double the notification, the notification UI is correct but two pings will triggered for each message that goes through the fastline flow (push from server -> fetch message directly -> notify message -> request sync -> sync finds new messages and notifies again) https://github.com/vector-im/element-android/blob/develop/vector/src/gplay/java/im/vector/app/gplay/push/fcm/VectorFirebaseMessagingService.kt#L174

The double ping will only happen if the sync triggered by the push returns _after_ the `getEvent/fastlane` , otherwise the event is ignored. The _fastlane_ flow is only entered when on wifi and we're not currently viewing the room the message is part of https://github.com/vector-im/element-android/blob/develop/vector/src/gplay/java/im/vector/app/gplay/push/fcm/VectorFirebaseMessagingService.kt#L187

**Comment by ouchadam** [comment]
ultimately my gut feeling is that we'll need to dynamic apply the `setOnlyAlertOnce` based on new content or updates/edits, I'm thinking we could check if the latest room message event is an update, and if so use `setOnlyAlertOnce(true)` to avoid the double ping 

there's a few changes needed to enable this

- We need to confirm that notifying twice with `setOnlyAlertOnce(false) setOnlyAlertOnce(true)` only pings once

- Add state to know if an existing event has been replaced https://github.com/vector-im/element-android/blob/develop/vector/src/main/java/im/vector/app/features/notifications/NotificationEventQueue.kt#L115

- When creating the room notifications https://github.com/vector-im/element-android/blob/develop/vector/src/main/java/im/vector/app/features/notifications/RoomGroupMessageCreator.kt#L69 make use of the new state, could be something like `setOnlyAlertOnce(events.last().isUpdate)`

File: changelog.d/46312.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Use the same notification rules than web / iOs client
```
Comment (ouchadam) [inline changelog.d/46312.misc:1]: is it worth calling out that the change will mean we will notify for all (new) messages?

**Review by ouchadam — APPROVED** [review]

File: changelog.d/46312.misc (hunk 1 above)
Comment (Claire1817) [reply changelog.d/46312.misc:1]: yes we can, i update the message

**Pull request merged**